web/src/app/core/resolvers/edit-project-token.resolver.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, Resolve, Router } from '@angular/router';
+import { of } from 'rxjs';
+import { catchError, switchMap, take } from 'rxjs/operators';
+
+// Dashboard hub model and services
+import { ProjectModel, ProjectTokenModel } from '../../shared/models/index.model';
+import { ProjectService } from '../services/index.service';
+import { TokenService } from '../services/token.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class EditProjectTokenResolver implements Resolve<ProjectTokenModel> {
+
+  constructor(
+    private tokenService: TokenService,
+    private router: Router
+  ) { }
+
+  resolve(route: ActivatedRouteSnapshot): any {
+    return this.tokenService.findOneById(route.params.projectUid, route.params.uid)
+      .pipe(
+        take(1),
+        catchError(() => {
+          this.router.navigate(['/']);
+          return of(new ProjectTokenModel());
```

Comment:
This should be handled in the component, the service might get reused somewhere else and they might not want the redirect

Reply:
Sorry @kykyk you were right here 🤦‍♂ . I didn't page attention that it was the resolver, I thought it was the service. Can it redirect to the project token list if token is not found